lib/datadog/appsec/contrib/faraday/patcher.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require_relative '../patcher'
+
+module Datadog
+  module AppSec
+    module Contrib
+      module Faraday
+        # Patcher for Faraday
+        module Patcher
+          include Datadog::AppSec::Contrib::Patcher
+
+          module_function
+
+          def patched?
+            Patcher.instance_variable_get(:@patched)
+          end
+
+          def target_version
+            Integration.version
+          end
+
+          def patch
+            require_relative 'middleware'
+            require_relative 'connection'
+
+            register_middleware!
+            add_default_middleware!
+
+            Patcher.instance_variable_set(:@patched, true)
+          end
+
+          def register_middleware!
+            ::Faraday::Middleware.register_middleware(datadog_appsec: Middleware)
+          end
+
+          def add_default_middleware!
```

Comment:
I think this name is misleading, what does it mean `default` in this case? Default faraday middleware or default something else?

Reply:
yes, you are right, let's call it `add_default_faraday_middleware`.
Context for this is - Faraday has a set of default middlewares that are used for all clients, and you can also add middlewares to each client separately.

Reply:
I would rephrase the verb then, we trying to configure (not add) the default faraday middlewares list, right?

Reply:
yes, let's call it `configure_default_faraday_connection`